main.js

```diff
@@ -0,0 +1,203 @@
+/*
+  Project Shift Week 1 Project - Tic-Tac-Toe
+  Austin Stevens
+*/
+
+/*
+  Players array to track both players
+  Each player has a name, character of X or O, and a status to indicate
+  whether he/she is able to play
+*/
+var TicTacToe = function() {
+  var board = [
+    ["1", "2", "3"],
+    ["4", "5", "6"],
+    ["7", "8", "9"]
+  ];
+  var currentPlayer = player1;
+  var gridSize = 3;
+  var moveCount = 0; // Tracking the number of turns taken.
+  var nameSet = false; // Indicates that setup is completed with valid names.
+
+  var player1 = {
+    name: '',
+    character: 'X'
+  };
+  var player2 = {
+    name: '',
+    character: 'O'
+  };
+
+  /*
+    Function to test if a player has won or a tie has occurred after each move.
+    The game ends if a win or tie occurs, and the result is logged to the console.
+  */
+  var checkWinner = function() {
+    var winChar = currentPlayer.character;
+    var winString = '';
+    var win1 = 'XXX';
+    var win2 = 'OOO';
+    var winCheck = function() {
+      if (winString == win1 || winString == win2) {
+        console.log(currentPlayer.name + ' wins!\nUse game.setup() to play again!');
+        reset();
+        return true;
+      }
+      return false;
+    };
+    // Check rows
+    for (var i = 0; i < gridSize; i++) {
+      winString = '';
+      for (var j = 0; j < gridSize; j++) {
+        if (board[i][j] == winChar) {
+          winString += winChar;
+        }
+        winCheck();
```

Comment:
This is a little confusing - having both a `checkWinner` and `winCheck` function. I get what's going on and I think having the `winCheck` helper is awesome, but I'd consider naming this differently.